camera_calibration/src/camera_calibration/calibrator.py

```diff
@@ -938,6 +939,35 @@ def pt2line(x0, y0, x1, y1, x2, y2):
         else:
             return None
 
+    def update_mask(self, img, screen, radius = 35):
+        """
+        :param img: source image
+        :type img: :class:`cvMat`
+
+        :param screen: display
+        :type screen: :class:`cvtColor`
+
+        :param radius: radius of the green circle which shows where corner detection has occurred
+        :type radius: :class:`int`
+
+        Update or if None create a red mask which shows where no corner detection has occurred
+        with the same size of screen and add on top of it detection areas displayed through
+        green circles which represents instead areas of the image where corners have been detected
+        """
+        height, width = img.shape[0], img.shape[1]
+        shape = screen.shape
+        scale_height, scale_width = height / shape[0], width / shape[1]
+        # Create mask if none
+        if self.mask is None:
+            self.mask = numpy.uint8(numpy.ones((shape[0], shape[1], 1)) * (0, 0, 255))
+        # Else update it 
+        else:
```

Comment:
why only do it in the else case? I think you always want to update the mask when this is called. This will return an empty mask for the first detection